docs/resources/data_record.md

```diff
@@ -0,0 +1,121 @@
+  environment_type = "Sandbox"
+  dataverse = {
+    language_code     = "1033"
+    currency_code     = "USD"
+    security_group_id = "00000000-0000-0000-0000-000000000000"
+  }
+}
+
+resource "powerplatform_data_record" "data_record_sample_contact1" {
+  environment_id     = powerplatform_environment.data_record_example_env.id
+  table_logical_name = "contact"
+  columns = {
+    firstname          = "John"
+    lastname           = "Doe"
+    telephone1         = "555-555-5555"
+    emailaddress1      = "johndoe@contoso.com"
+    anniversary        = "2024-04-10"
+    annualincome       = 1234.56
+    birthdate          = "2024-04-10"
+    description        = "This is the description of the the terraform \n\nsample contact"
+  }
+}
+
+resource "powerplatform_data_record" "data_record_sample_contact2" {
+  environment_id     = powerplatform_environment.data_record_example_env.id
+  table_logical_name = "contact"
+  columns = {
+    firstname          = "Jane"
+    lastname           = "Doe"
+    telephone1         = "555-555-5555"
+    emailaddress1      = "janedoe@contoso.com"
+    anniversary        = "2024-04-11"
+    annualincome       = 1234.56
+    birthdate          = "2024-04-11"
+    description        = "This is the description of the the terraform \n\nsample contact"
+  }
+}
+
+resource "powerplatform_data_record" "data_record_accounts" {
+  environment_id     = powerplatform_environment.data_record_example_env.id
+  table_logical_name = "account"
+  columns = {
+    name                = "Sample Account"
+    creditonhold        = false
+    address1_latitude   = 47.63958
+    description         = "This is the description of the sample account"
+    revenue             = 5000000
+    accountcategorycode = 1
+
+    primarycontactid = { # one to many
+      table_logical_name = powerplatform_data_record.data_record_sample_contact1.table_logical_name
+      data_record_id      = powerplatform_data_record.data_record_sample_contact1.id
+    }
+
+    contact_customer_accounts = toset([ # many to one
+      {
+        table_logical_name = powerplatform_data_record.data_record_sample_contact1.table_logical_name
+        data_record_id      = powerplatform_data_record.data_record_sample_contact1.id
+      },
+      {
+        table_logical_name = powerplatform_data_record.data_record_sample_contact2.table_logical_name
+        data_record_id      = powerplatform_data_record.data_record_sample_contact2.id
+      }
+    ])
+
+    cr4d0_Account_Contact_Contact = toset([ # many to many
+      {
+        table_logical_name = powerplatform_data_record.data_record_sample_contact1.table_logical_name
+        data_record_id      = powerplatform_data_record.data_record_sample_contact1.id
+      },
+      {
+        table_logical_name = powerplatform_data_record.data_record_sample_contact2.table_logical_name
+        data_record_id      = powerplatform_data_record.data_record_sample_contact2.id
+      }
+    ])
+  }
+}
+```
```

Comment:
Our example for documentation should not be account/contact.  We should use something like business unit or other configuration table.  We do not want people thinking that they should use this for business data.

Reply:
I'd like to get your input here, let's discuss it offline

Reply:
@polatengin you can use this, just add the variables:

```
resource "powerplatform_data_record" "role" {
  environment_id     = var.environment_id
  table_logical_name = "role"

  columns = {
    name = "my custom role"

    businessunitid = {
      table_logical_name = "businessunit"
      data_record_id     = var.parent_business_unit_id
    }
  }
}

resource "powerplatform_data_record" "team" {
  environment_id     = var.environment_id
  table_logical_name = "team"
  columns = {
    name        = "main team"
    description = "main team description"

    teamroles_association = [
      {
        table_logical_name = "role"
        data_record_id     = powerplatform_data_record.role.id
      }
    ]
  }
}
```

Reply:
I've updated the samples file